lib/FtpConnection.js

```diff
@@ -1074,12 +1074,20 @@ FtpConnection.prototype._command_RNTO = function(commandArg) {
   var self = this;
   var fileto = withCwd(self.cwd, commandArg);
   self.fs.rename(pathModule.join(self.root, self.filefrom), pathModule.join(self.root, fileto), function(err) {
+    var status = 'success';
     if (err) {
+      status = 'error';
       self._logIf(LOG.ERROR, 'Error renaming file from ' + self.filefrom + ' to ' + fileto);
       self.respond('550 Rename failed' + (err.code === 'ENOENT' ? '; file does not exist' : ''));
     } else {
       self.respond('250 File renamed successfully');
     }
+    // Adding event emitter for completed file renaming.
+    self.emit('file:rnto', status, {
+      fileFrom: self.filefrom,
+      fileTo: fileto,
+      error: err
```

Comment:
I'd prefer an additional `errorState: Boolean(err)` like we have with the `file:stor` event, and `error: (err || false)` as a convenience to users to make property access (e.g. `err.name`, `err.message`) always safe.

Reply:
Ok. Doing it. One extra question: Which are the most recent compatible node version for this project?

Reply:
I've no idea but I expect it to work with all Node.js LTS versions.